flang/lib/Lower/FIRBuilder.cpp

```diff
@@ -434,6 +441,49 @@ struct CharacterOpsBuilderImpl {
 
     return {substringRef, substringLen};
   }
+
+  mlir::Value createLenTrim(Char str) {
+    // Note: Runtime for LEN_TRIM should also be available at some
+    // point. For now use an inlined implementation.
+    auto indexType = builder.getIndexType();
+    auto i1Type = mlir::IntegerType::get(1, builder.getContext());
+    mlir::Value len = builder.createHere<fir::ConvertOp>(indexType, str.len);
+    auto data = materializeValue(str);
+    auto one = builder.createIntegerConstant(indexType, 1);
+    auto minusOne = builder.createIntegerConstant(indexType, -1);
+    auto zero = builder.createIntegerConstant(indexType, 0);
+    auto trueVal = builder.createIntegerConstant(i1Type, 1);
+    auto blank = createBlankConstant(str.getCharacterType());
+    // FIXME: Conversion wrong in general, it would be cool to compare fir.char
+    // without cast. Needs the kindmap.
+    auto intType = mlir::IntegerType::get(8, builder.getContext());
+    blank = builder.createHere<fir::ConvertOp>(intType, blank);
+    // initVal must not be random to account for the zero length case where the
+    // no iteration of the loop is ever executed.
+    mlir::Value initVal = builder.createHere<mlir::SubIOp>(len, one);
+
+    auto iterWhile =
+        builder.createHere<fir::IterWhileOp>(initVal, zero, minusOne, trueVal, initVal);
```

Comment:
This line might need reformatting to 80 columns.

Reply:
Fixed, thanks.